File: ios/Runner/Config/NetmeraEnvironment.swift
///
/// Copyright (c) 2026 Netmera Research.
///

import Foundation

enum NetmeraSettingsKeys {
    static let environment = "environment"
    static let baseURL = "baseURL"
    static let APIKey = "APIKey"
}

enum NetmeraEnvironment: String {
    case test
    case preprod
    case prod
    case custom

    var url: String {
        switch self {
        case .test: return "https://sdk-cloud-test.sdpaas.com"
        case .preprod: return "https://sdk-cloud-uat.sdpaas.com"
        case .prod: return "https://sdkapi.netmera.com"
        case .custom: return ""
        }
    }

    var defaultApiKey: String {
        switch self {
        case .test: return "gFtyH_nz5WCdXraTsOOgL25er1sBpuQd2XKJi23QV17ebnzYFWCHb1x6Q7ILiK_bwlZ-FPtUSzU"
        case .preprod: return "gFtyH_nz5WCdXraTsOOgL25er1sBpuQdFa8bAMVGhv9Xo5voVRPFY2gFcVkuTQeZIgoa3gN1rww"
        case .prod: return "gFtyH_nz5WCdXraTsOOgL25er1sBpuQdYkuURoDYBA5tcjTUZqlWw0M4LNxpP9V9rH0FOnONCGM"
        case .custom: return ""
        }
    }
}

File: android/app/src/main/kotlin/com/netmera/demo/flutter/config/NetmeraEnvironment.kt
///
/// Copyright (c) 2026 Netmera Research.
///

package com.netmera.demo.flutter.config

enum class NetmeraEnvironment(
    val key: String,
    val url: String,
    val defaultApiKey: String
) {
    TEST(
        "test",
        "https://sdk-cloud-test.sdpaas.com",
        "gFtyH_nz5WCdXraTsOOgL25er1sBpuQd2XKJi23QV17ebnzYFWCHb1x6Q7ILiK_bwlZ-FPtUSzU"
    ),
    PREPROD(
        "preprod",
        "https://sdk-cloud-uat.sdpaas.com",
        "gFtyH_nz5WCdXraTsOOgL25er1sBpuQdFa8bAMVGhv9Xo5voVRPFY2gFcVkuTQeZIgoa3gN1rww"
    ),
    PROD(
        "prod",
        "https://sdkapi.netmera.com",
        "gFtyH_nz5WCdXraTsOOgL25er1sBpuQdYkuURoDYBA5tcjTUZqlWw0M4LNxpP9V9rH0FOnONCGM"
    ),
    CUSTOM("custom", "", "");

    val displayName: String
        get() = when (this) {
            TEST -> "Test"
            PREPROD -> "Pre-prod"
            PROD -> "Prod"
            CUSTOM -> "Custom"
        }

    companion object {
        fun fromKey(key: String?): NetmeraEnvironment =
            entries.find { it.key == key } ?: PREPROD
    }
}

